neo/Plugins/Plugin.cs

```diff
@@ -62,6 +62,10 @@ protected virtual void OnPluginsLoaded()
         {
         }
 
+        protected virtual void OnShutdown()
```

Comment:
Should all plugins implement `IDisposable`?

Reply:
l think that could achieve the same.

Reply:
Do we want to enforce plugins to implement `Dispose()` or provide a default implementation they optionally override if needed?

1. `public abstract void Dispose();`
2. `public void Dispose() {};`

Reply:
`virtual` is good.